storage/aptosdb/src/pruner/utils.rs

```diff
@@ -18,18 +19,27 @@ use std::{sync::Arc, time::Instant};
 pub fn create_db_pruners(
     ledger_db: Arc<DB>,
     state_merkle_db: Arc<DB>,
-) -> Vec<Mutex<Arc<dyn DBPruner + Send + Sync>>> {
+    storage_pruner_config: StoragePrunerConfig,
+) -> Vec<Option<Mutex<Arc<dyn DBPruner + Send + Sync>>>> {
     vec![
-        Mutex::new(Arc::new(StateStorePruner::new(
-            Arc::clone(&state_merkle_db),
-            0,
-            Instant::now(),
-        ))),
-        Mutex::new(Arc::new(LedgerPruner::new(
-            Arc::clone(&ledger_db),
-            Arc::new(TransactionStore::new(Arc::clone(&ledger_db))),
-            Arc::new(EventStore::new(Arc::clone(&ledger_db))),
-            Arc::new(LedgerStore::new(Arc::clone(&ledger_db))),
-        ))),
+        if storage_pruner_config.state_store_prune_window.is_some() {
```

Comment:
Same as above

Reply:
Discussed offline.

if using map, we need to cast the pruner to its trait and will need to make it a Box

 ```
25 | /         storage_pruner_config.state_store_prune_window.map(|_|
26 | | Mutex::new(Arc::new(StateStorePruner::new(
27 | |                 Arc::clone(&state_merkle_db),
28 | |                 0,
29 | |                 Instant::now(),
30 | |             )))),
   | |________________^ expected trait object `dyn DBPruner`, found struct `StateStorePruner`
   |
   = note: expected enum `std::option::Option<aptos_infallible::Mutex<std::sync::Arc<(dyn DBPruner + Send + Sync + 'static)>>>`
              found enum `std::option::Option<aptos_infallible::Mutex<std::sync::Arc<StateStorePruner>>>`
```
So will keep it as is.